specification/computeschedule/resource-manager/Microsoft.ComputeSchedule/preview/2024-06-01-preview/computeschedule.json

```diff
@@ -0,0 +1,1348 @@
+        "Scheduled",
+        "PendingExecution",
+        "Executing",
+        "Succeeded",
+        "Failed",
+        "Cancelled",
+        "Blocked"
+      ],
+      "x-ms-enum": {
+        "name": "OperationState",
+        "modelAsString": true,
+        "values": [
+          {
+            "name": "Unknown",
+            "value": "Unknown",
+            "description": "The default value for the operation state enum"
+          },
+          {
+            "name": "PendingScheduling",
+            "value": "PendingScheduling",
+            "description": "Operations that are pending scheduling"
+          },
+          {
+            "name": "Scheduled",
+            "value": "Scheduled",
+            "description": "Operations that have been scheduled"
+          },
+          {
+            "name": "PendingExecution",
+            "value": "PendingExecution",
+            "description": "Operations that are waiting to be executed"
+          },
+          {
+            "name": "Executing",
+            "value": "Executing",
+            "description": "Operations that are in the process of being executed"
+          },
+          {
+            "name": "Succeeded",
+            "value": "Succeeded",
+            "description": "Operations that suceeded"
+          },
+          {
+            "name": "Failed",
+            "value": "Failed",
+            "description": "Operations that have failed"
+          },
+          {
+            "name": "Cancelled",
+            "value": "Cancelled",
+            "description": "Operations that have been cancelled by the user"
+          },
+          {
+            "name": "Blocked",
+            "value": "Blocked",
+            "description": "Operations that are blocked"
+          }
+        ]
+      }
+    },
+    "OptimizationPreference": {
+      "type": "string",
+      "description": "The preferences customers can select to optimize their requests to ScheduledActions",
+      "enum": [
+        "Cost",
+        "Availability",
+        "CostAvailabilityBalanced"
+      ],
+      "x-ms-enum": {
+        "name": "OptimizationPreference",
+        "modelAsString": true,
+        "values": [
+          {
+            "name": "Cost",
+            "value": "Cost",
+            "description": "Optimize while considering cost savings"
+          },
+          {
+            "name": "Availability",
+            "value": "Availability",
+            "description": "Optimize while considering availability of resources"
+          },
+          {
+            "name": "CostAvailabilityBalanced",
+            "value": "CostAvailabilityBalanced",
+            "description": "Optimize while considering a balance of cost and availability"
+          }
+        ]
+      }
+    },
+    "ResourceOperation": {
+      "type": "object",
+      "description": "High level response from an operation on a resource",
+      "properties": {
+        "resourceId": {
+          "type": "string",
```

Comment:
>type [](http://example.com/codeflow?start=11&length=4)

is this an ArmId always? If so please add "format": "armId"

Reply:
Quick question on this, will this require a new version since this is a breaking change according to [this document](https://microsoft.sharepoint.com/:w:/r/teams/azureresourcemanagerteam/_layouts/15/Doc.aspx?sourcedoc=%7B0D01EC65-C772-4A6E-9326-0279E586520C%7D&file=RestAPI_BreakingChangeGuide.docx&action=default&mobileredirect=true&DefaultItemOpen=1&share=IQFl7AENcsduSpMmAnnlhlIMAWyykwWAcYcItf7Ewp6Z6qY)?

Reply:
changed this format to armid